databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,291 @@
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all the scopes')
+@click.option('--output', default=None, help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_scopes(api_client, output):
+    """
+    Lists all secret scopes available in the workspace.
+    """
+    scopes_json = SecretApi(api_client).list_scopes()
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(scopes_json))
+    else:
+        click.echo(tabulate(_scopes_to_table(scopes_json), headers=SCOPE_HEADER))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a scope')
+@click.option('--scope', required=True)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_scope(api_client, scope):
+    """
+    Deletes a secret scope.
+    """
+    SecretApi(api_client).delete_scope(scope)
+
+
+def verify_and_read_value(string_value, bytes_value, value, no_strip):
+    """
+    Verify exactly one among the options is set to True and translate the value
+    """
+    valid_count = 0
+    for v in [string_value, bytes_value]:
+        if v:
+            valid_count += 1
+
+    if valid_count != 1:
+        error_and_quit("You must specify exactly one in options"
+                       " [--string-value, --bytes-value]")
+
+    if string_value:
+        return translate_value(value, no_strip), None
+    if bytes_value:
+        return None, base64.encodestring(translate_value(value, no_strip))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Writes a secret')
+@click.option('--scope', required=True)
+@click.option('--key', required=True)
+@click.option('--string-value', is_flag=True, default=False,
+              help='A flag indicates the value will be stored in UTF-8 (MB4) form')
+@click.option('--bytes-value', is_flag=True, default=False,
+              help='A flag indicates the value will be store as bytes')
+@click.option('--no-strip', is_flag=True, default=False,
+              help='Indicate the trailing \'\\n\' should not be stripped from input value')
+@click.argument('VALUE', required=False)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def write_secret(api_client, scope, key, string_value, bytes_value, no_strip, value):
+    """
+    Inserts a secret under the provided scope with the given name. Overwrite if the name exists.
+
+    You should specify exactly one flag to indicate the value is "string-value" or "bytes-value".
+
+    If VALUE is an empty string or not provided, an editor will be opened for you to enter your
+    secret value.
+    If VALUE starts with '@', the rest of the string will be seen as a path to a file, the content
+    of file will be read as content.
+    Otherwise, the VALUE itself will be seen as secret value.
+
+    The specification for input can be found at:
```

Comment:
I think we can remove this line, because you've already explained the input requirements above. Also, we don't necessarily want to point them to Azure only documentation, since we'll have to change this when we support AWS :).